client/exec.go

```diff
@@ -0,0 +1,305 @@
+// Copyright (c) 2021 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/gorilla/websocket"
+
+	"github.com/canonical/pebble/internal/wsutil"
+)
+
+// ExecOptions are the main options for the Exec call.
+type ExecOptions struct {
+	// Required: command and arguments (first element is the executable)
+	Command []string
+
+	// Optional environment variables
+	Environment map[string]string
+
+	// Optional working directory (default is $HOME or "/" if $HOME not set).
+	WorkingDir string
```

Comment:
Hmm, it's hard to tell. Docker calls the directive `WORKDIR`, but that's more because they have very short directive names like `COPY`, `ADD`, etc. In their [docs](https://docs.docker.com/engine/reference/builder/#workdir) when spelled out it's the "working directory". Systemd [calls](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#WorkingDirectory=) the setting `WorkingDirectory` in full. Go exec calls it just `Dir`, and Python subprocess calls it `cwd`. I think that's 4 different styles. In any case, I think in full it's the "working directory", so `WorkingDir` seemed right to me. `WorkingDirectory` would work (excuse the pun) too, but it's a bit long.

Reply:
Okay, thanks for checking.